x-pack/legacy/plugins/lens/public/metric_visualization_plugin/auto_scale.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+interface Props {
+  children: React.ReactNode | React.ReactNode[];
+}
+
+interface State {
+  scale: number;
+}
+
+export class AutoScale extends React.Component<Props, State> {
```

Comment:
Is there a specific reason we are not using hooks here? As far as I can see, we are using function components and hooks everywhere else. No strong feelings about this, just curious.

Reply:
Yes. I used hooks at first, and it kept getting into infinite loops that were hard to track down. Things were simplified significantly when I moved to a class.